controller/users.go

```diff
@@ -361,9 +381,55 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 			}
 		}
 	}
+	c.notifyWITService(ctx, ctx.RequestData) // TODO: Dont ignore error
 	return returnResponse
 }
 
+// addDelegationFlag adds information for WIT service to know that
+// this was forwarded from AUTH
+func addDelegationFlag(ctx *app.UpdateUsersContext) *app.UpdateUsersContext {
+	ctx.Context = context.WithValue(ctx.Context, delegationFlag, true)
+	return ctx
+}
+
+func (c *UsersController) notifyWITService(ctx *app.UpdateUsersContext, request *goa.RequestData) error {
+	remoteWITClient, err := c.createRemoteWITClient(ctx.Context, request)
+	if err != nil {
+		return err
+	}
+	updateUserPayload := &witservice.UpdateUsersPayload{
+		Data: &witservice.UpdateUserData{
+			Attributes: &witservice.UpdateIdentityDataAttributes{
+				Bio:                   ctx.Payload.Data.Attributes.Bio,
+				Company:               ctx.Payload.Data.Attributes.Company,
+				ContextInformation:    ctx.Payload.Data.Attributes.ContextInformation,
+				Email:                 ctx.Payload.Data.Attributes.Email,
+				FullName:              ctx.Payload.Data.Attributes.FullName,
+				ImageURL:              ctx.Payload.Data.Attributes.ImageURL,
+				RegistrationCompleted: ctx.Payload.Data.Attributes.RegistrationCompleted,
+				URL:      ctx.Payload.Data.Attributes.URL,
+				Username: ctx.Payload.Data.Attributes.Username,
+			},
+			Type: ctx.Payload.Data.Type,
+		},
+	}
+	ctx = addDelegationFlag(ctx)
+	_, err = remoteWITClient.UpdateUsers(goasupport.ForwardContextRequestID(ctx), "", updateUserPayload)
+	return err
+}
+
+func (c *UsersController) createRemoteWITClient(ctx context.Context, request *goa.RequestData) (*witservice.Client, error) {
+	witEndpoint, err := c.config.GetWITEndpoint(request)
+	if err != nil {
+		return nil, err
+	}
+	WITClient, err := remoteservice.CreateSecureRemoteWITClient(ctx, witEndpoint, nil)
```

Comment:
Here is my suggestion on how we could get the service account token to sign the request.

Let's add a new method to token/token.go or tokencontex/token_context.go (preferable but it may cause cycle dependencies):

```
func ReadManagerFromContext(ctx context.Context) (*Manager, error) {
	tm := tokencontext.ReadTokenManagerFromContext(ctx)
	if tm == nil {
		log.Error(ctx, map[string]interface{}{
			"token": tm,
		}, "missing token manager")

		return nil, errs.New("Missing token manager")
	}
	return tm.(Manager), nil
}

```

then we can use the following code to get the Service Account token:

```
manager, err := ReadManagerFromContext(ctx)
token, err := manager.ServiceAccount(request)

```

Reply:
When I pass on the User Update payload to the WIT Rest API , the following piece of code determines the ID of the user 

```
func (mgm *tokenManager) Locate(ctx context.Context) (uuid.UUID, error) {
	token := goajwt.ContextJWT(ctx)
	if token == nil {
		return uuid.UUID{}, errors.New("Missing token") // TODO, make specific tokenErrors
	}ContextIdent
	id := token.Claims.(jwt.MapClaims)["sub"]
	if id == nil {
		return uuid.UUID{}, errors.New("Missing sub")
	}
	idTyped, err := uuid.FromString(id.(string))
	if err != nil {
		return uuid.UUID{}, errors.New("uuid not of type string")
	}
	return idTyped, nil
}
```

In case of using the service account token, the ID here would be the token service account and not that of the user.

Reply:
Well, after we finish this migration we won't need to have any rest API on WIT side to update users based on user's access token. Only Auth will be using such API. So, it makes sense to do whatever is needed to support it on WIT side (new endpoint to update/create users OR refactor the existing endpoint(s) to support required payloads or user IDs or whatever). The case when the user is updated via user's access token is becoming a temporal thing which we will remove after finish the migration.

Reply:
Thanks I agree that most of it is temporal. but -

It's not entirely temporal because user update from AUTH to WIT would keep on happening for good.. since WIT would need to be aware of user details forever. 

Here's what I'm proposing. https://github.com/fabric8-services/fabric8-wit/pull/1635

Reply:
Sure. I'm talking about scenario when WIT is called by UI/user/whatever to update the user profile and the user's access token is passed. We need to support it for now but we will remove it when merge is done.
The only scenario we will need to support in long term is updating user in WIT by Auth. So this API should be solid.